### PR TITLE
Pull testing framework into OpenMLS crate

### PR DESCRIPTION
**Pull request by kkohbrok**
This PR pulls the testing framework from the `test_utils` crate into the `test_utils` module inside the OpenMLS crate. It also moves the tree kem test vector generation to the same place where tree kem test vectors are tested. This will allow us to move the interop client to main and start doing interop from the main branch

### CODE REVIEW

File: openmls/src/lib.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ mod codec;
 
 #[cfg(any(feature = "expose-test-vectors", test))]
 #[macro_use]
-pub mod test_util;
+pub mod test_utils;
```
Comment (franziskuskiefer) [inline openmls/src/lib.rs:34]: Can you change the feature to something like `test-utils`?

File: openmls/src/test_utils/mod.rs [hunk 2]
```diff
@@ -7,6 +7,8 @@ use std::{
     io::{BufReader, Write},
 };
 
+pub mod test_framework;
+
 pub fn write(file_name: &str, obj: impl Serialize) {
```
Comment (franziskuskiefer) [inline openmls/src/test_utils/mod.rs:12]: A  little unrelated, but can you check that all these functions in here need to be public?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
lgtm with the small nit

File: openmls/src/lib.rs (hunk 1 above)
Comment (kkohbrok) [reply openmls/src/lib.rs:34]: Done.

File: openmls/src/test_utils/mod.rs (hunk 2 above)
Comment (kkohbrok) [reply openmls/src/test_utils/mod.rs]: Done. All of these functions are crate-public now.

File: openmls/Cargo.toml [hunk 3]
```diff
@@ -27,6 +27,7 @@ itertools = {version = "0.10", optional = true}
 default = ["rust-crypto"]
 rust-crypto = ["evercrypt/rust-crypto-aes"]
 expose-test-vectors = ["itertools"]
+test_utils = []
```
Comment (franziskuskiefer) [inline openmls/Cargo.toml:30]: There's no clear guideline on this yet ([see for example this discussion](https://github.com/rust-lang/api-guidelines/discussions/101)). But all other features are kebab case. So we should probably stick to it

File: openmls/src/lib.rs [hunk 4]
```diff
@@ -29,7 +29,7 @@ pub mod error;
 #[macro_use]
 mod codec;
 
-#[cfg(any(feature = "expose-test-vectors", test))]
+#[cfg(any(feature = "test-utils", feature = "expose-test-vectors", test))]
```
Comment (franziskuskiefer) [inline openmls/src/lib.rs:32]: you even use kebab case here 😉

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]

File: openmls/Cargo.toml (hunk 3 above)
Comment (kkohbrok) [reply openmls/Cargo.toml:30]: Right you are. That was really just a typo.

File: openmls/Cargo.toml [hunk 5]
```diff
@@ -27,6 +27,7 @@ itertools = {version = "0.10", optional = true}
 default = ["rust-crypto"]
 rust-crypto = ["evercrypt/rust-crypto-aes"]
 expose-test-vectors = ["itertools"]
```
Comment (franziskuskiefer) [inline openmls/Cargo.toml:29]: Do you want to keep both feature flags? I thought `test-utils` could expose the test vectors as well.

Comment (kkohbrok) [reply openmls/Cargo.toml:29]: I thought that was what you were suggesting. I have no preference either way. Then I would propose we only keep the `expose-test-vector` for now or at least until we need to expose only the test_utils.

Comment (kkohbrok) [reply openmls/Cargo.toml:29]: Same as above, only that I actually changed the name everywhere to `test-utils`, so there's no more `expose-test-vectors`.

**Review by franziskuskiefer — APPROVED** [review]

**Review by raphaelrobert — APPROVED** [review]
Looks straight forward and relying on @franziskuskiefer for the nits.

**Pull request merged**